internal/provider/models/shared_source.go

```diff
@@ -0,0 +1,92 @@
+package models
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	. "github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	. "github.com/mezmo/terraform-provider-mezmo/internal/client"
+)
+
+var PUSH_SOURCE_TYPES = []string{
+	"http",
+	"splunk-hec",
+	"kinesis-firehose",
+	"fluent",
+	"logstash",
+	"mezmo-agent",
+	"mezmo-datadog-source",
+	"webhook",
+	"prometheus-remote-write",
+	"open-telemetry-metrics",
+	"open-telemetry-logs",
+	"open-telemetry-traces",
+}
```

Comment:
We do have these strings defined as const variables in the source model code base. Those const variables is what the `New___Resource()` functions use ([example](https://github.com/mezmo/terraform-provider-mezmo/blob/main/internal/provider/source_resource_definitions.go#L24)). Maybe we should use those here to avoid the risk of value drift?

Reply:
Good call, I can look into that

Reply:
No go.  The problem is that `ingress_type` is statically defined on the class.  We don't track that from within this provider.  This list was manually created for all push sources (excluding things like s3).